### ISSUE TITLE
GPU_PrimitiveBatch issue

### ISSUE THREAD

**Issue by kirilledelman** [grimfang4/sdl-gpu#115]
Jonathan,

I'm having trouble using `GPU_PrimitiveBatch`. My intention is to draw a series of line segments with current color. I make a call with:

`GPU_PrimitiveBatch( NULL, target, GPU_LINE_STRIP, 5, floatPointsPackedAsXY, 5, uShortIndexes, GPU_BATCH_XY )`

and it crashes in `get_complete_mod_color`, because I'm simultaneously not supplying image, and not using colors. I don't need per-vertex color, or texture.

Please hlep.

**Comment by kirilledelman** [comment]
By stepping through code, and temporarily patching `get_complete_mod_color` I arrived to actual rendering inside `PrimitiveBatchV` in `Renderer_GL_Common.inl`:
```
if(indices == NULL)
            glDrawArrays(GL_TRIANGLES, 0, num_indices);
        else
            glDrawElements(GL_TRIANGLES, num_indices, GL_UNSIGNED_SHORT, (void*)0);
```
As you see, it's always drawing triangles, not the specified primitive. Is this simply unfinished? I'm going to try and plug in the primitive from parameter here. I assume this should be fixed in main branch.

**Comment by kirilledelman** [comment]
For my purposes, it would be perfectly acceptable to use `GPU_Polygon`, to draw a line strip (with thickness), if `GPU_Polygon` had another parameter that let me specify if I want to close this polygon, or keep it open (line strip). 👍

**Comment by kirilledelman** [comment]
Could you take a look at my commit, please?

**Comment by grimfang4** [comment]
Yep, these are good thoughts.  Thanks for your work on this!  (other discussion on merge thread)
